File: Java/secondLargest.java
public class secondLargest {
    public static void main(String[]agrs) {
        int numbers[] = {1, 2, 3, 3, 4, 9, 2, 7, 3, 8};
        int firstLargest = 0, secondLargest=-1;

        for (int i: numbers){
            if (firstLargest < i) {
                secondLargest = firstLargest;
                firstLargest = i;
            }
            else if (secondLargest < i)
                secondLargest = i;
        }

        System.out.println(secondLargest);
    }
}

File: Python/secondLargest.py
numbers = [1, 2, 3, 3, 4, 5, 2, 7, 3, 8]
max1 = 0
max2 = -1

for n in numbers:

    if n > max1:
        max2 = max1
        max1 = n

    elif n > max2:
        max2 = n

print(max2)